spring-boot-project/spring-boot-test-autoconfigure/src/main/java/org/springframework/boot/test/autoconfigure/hateoas/HypermediaTestAutoConfiguration.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.boot.test.autoconfigure.hateoas;
+
+import java.util.List;
+
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.http.codec.CodecsAutoConfiguration;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.web.codec.CodecCustomizer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.hateoas.config.HypermediaMappingInformation;
+import org.springframework.http.codec.json.Jackson2JsonDecoder;
+import org.springframework.http.codec.json.Jackson2JsonEncoder;
+import org.springframework.test.web.reactive.server.WebTestClient;
+import org.springframework.util.Assert;
+import org.springframework.util.MimeType;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * @author Greg Turnquist
+ */
+@Configuration(proxyBeanMethods = false)
+@AutoConfigureAfter({CodecsAutoConfiguration.class})
+@EnableConfigurationProperties
+public class HypermediaTestAutoConfiguration {
+
+	@Bean
+	@ConditionalOnMissingBean
```

Comment:
This will back off if any other `CodecCustomizer` beans exist. I suspect that isn't what you want.

Reply:
Certainly not. 

I wanted to guard against the other branch that introduces `WebClient` support (#20374) and basically has the same type of bean. Just in another module.

Suggestions? Perhaps drop the backing off policy here and now, and then resolve such bean conflicts when we get to #20374?

Reply:
The `CodecCustomizer` here and in #20374 seem to be identical. I suspect it could be defined in one place. If so, it should probably be in `spring-boot-autoconfigure`. It could then be included via `spring.factories` in the relevant test auto-configuration as well.

Reply:
I guess I seemed to have issues (initially) with `spring-boot-test-autoconfigure` seeing things so I copied it.

Let me polish up #20374 applying some of these comments there. After that is merged, I can rebase this to use that.